src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamWriteAsyncMemoryOverloadsTests.cs

```diff
@@ -385,378 +428,284 @@ End Class
-using System;
-using System.IO;
-using System.Threading;
-class C
-{
-    public async void M()
-    {
-        using (FileStream fs = new FileStream(""path.txt"", FileMode.Create))
-        {
-            await fs.WriteAsync(new byte[]{ 0xBA, 0x5E, 0xBA, 0x11, 0xF0, 0x07, 0xBA, 0x11 }, 0, 8, new CancellationToken());
-        }
-    }
-}
-            ", GetCSResult(11, 19, 11, 125));
-        }
+        #endregion
 
-        [Fact]
-        public Task CS_Analyzer_Diagnostic_ConfigureAwait()
-        {
-            return AnalyzeCSAsync(@"
-using System;
-using System.IO;
-using System.Threading;
-class C
-{
-    public async void M()
-    {
-        byte[] buffer = { 0xBA, 0x5E, 0xBA, 0x11, 0xF0, 0x07, 0xBA, 0x11 };
-        using (FileStream fs = new FileStream(""path.txt"", FileMode.Create))
-        {
-            await fs.WriteAsync(buffer, 0, buffer.Length).ConfigureAwait(false);
-        }
-    }
-}
-            ", GetCSResult(12, 19, 12, 80));
-        }
+        #region C# - Diagnostic - Analyzer
 
-        [Fact]
-        public Task CS_Analyzer_Diagnostic_CancellationToken_ConfigureAwait()
-        {
-            return AnalyzeCSAsync(@"
+        [Theory]
+        [InlineData("buffer, 0, buffer.Length",
+                    "buffer.AsMemory(0, buffer.Length)",
+                    false, 57)]
+        [InlineData("buffer, 0, buffer.Length",
+                    "buffer.AsMemory(0, buffer.Length)",
+                    true, 57)]
+        [InlineData("buffer, 0, buffer.Length, new CancellationToken()",
+                    "buffer.AsMemory(0, buffer.Length), new CancellationToken()",
+                    false, 82)]
+        [InlineData("buffer, 0, buffer.Length, new CancellationToken()",
+                    "buffer.AsMemory(0, buffer.Length), new CancellationToken()",
+                    true, 82)]
+        public Task CS_Analyzer_Diagnostic_VarByteArray(string originalArgs, string fixedArgs, bool withConfigureAwait, int endColumn)
+        {
+            string source = @"
 using System;
 using System.IO;
 using System.Threading;
 class C
-{
+{{
     public async void M()
-    {
-        byte[] buffer = { 0xBA, 0x5E, 0xBA, 0x11, 0xF0, 0x07, 0xBA, 0x11 };
-        using (FileStream fs = new FileStream(""path.txt"", FileMode.Create))
-        {
-            await fs.WriteAsync(buffer, 0, buffer.Length, new CancellationToken()).ConfigureAwait(false);
-        }
-    }
-}
-            ", GetCSResult(12, 19, 12, 105));
-        }
-
-        [Fact]
-        public Task CS_Analyzer_Diagnostic_InlineBuffer_ConfigureAwait()
-        {
-            return AnalyzeCSAsync(@"
+    {{
+        using (FileStream s = new FileStream(""path.txt"", FileMode.Create))
+        {{
+            byte[] buffer = {{ 0xBA, 0x5E, 0xBA, 0x11, 0xF0, 0x07, 0xBA, 0x11 }};
+            await {0};
+        }}
+    }}
+}}
+            ";
+            return GetCSWriteAsyncDiagnostic(source, originalArgs, fixedArgs, withConfigureAwait, 12, 19, 12, endColumn);
```

Comment:
Same suggestion here about naming. Either use `CSharpVerifyCodeFixAsync` or `VerifyCSharpCodeFixAsync` to clarify the actual method semantics.

Reply:
Dibe

Reply:
Done